hail/src/main/scala/is/hail/expr/ir/PCode.scala

```diff
@@ -14,9 +14,9 @@ object PCode {
     case pt: PCanonicalArray =>
       new PCanonicalIndexableCode(pt, coerce[Long](code))
     case pt: PCanonicalSet =>
-      new PCanonicalIndexableCode(pt, coerce[Long](code))
+      new PCanonicalIndexableCode(pt.arrayRep, coerce[Long](code))
```

Comment:
You can't do this.  PCode ptype must match the IR ptype.  I think you can make the pcindexable code carry an arraybackend container.